snapcraft/internal/meta/snap.py

```diff
@@ -210,6 +239,13 @@ def from_dict(cls, snap_dict: Dict[str, Any]) -> "Snap":
         if "adopt-info" in snap_dict:
             snap.adopt_info = snap_dict["adopt-info"]
 
+        # Set grade to "stable" unless building from an non-stable base.
+        # We do not want snaps built from a non stable risk to have a stable
+        # grade as this base might still suffer ABI or API changes.
+        snap.grade = _get_grade_for_snap(
```

Comment:
What do you think about:
1. set `grade: str = "stable"` in `__init__()` so the default is always set (if one were to create a Snap not from dict)
2. which then lets allows grade to be moved into `_MANDATORY_PACKAGE_KEYS`
3. decoupling validation/warning logic into `validate_grade()`, called by `validate()`
4. with that decoupled, it's then something like:
```
if "grade" not in snap_dict:
    snap.grade = _get_base_grade(base)
```

Reply:
we should have a conversation again, I don't think this is possible